lib/datadog/core/configuration/components.rb

```diff
@@ -99,6 +99,8 @@ def initialize(settings)
           @health_metrics = self.class.build_health_metrics(settings)
           @telemetry = self.class.build_telemetry(settings, agent_settings, logger)
           @appsec = Datadog::AppSec::Component.build_appsec_component(settings)
+
+          self.class.configure_tracing(settings)
```

Comment:
I left this separate because `tracing` is not `tracer`.
The `build_tracer` method explicitly does the work to return a tracer instance.

The new method `configure_tracing` does work to configure any generic needs for the tracing product.
Ideally, the `tracing` product can configure itself without explicitly having to return a tracer instance, that is then managed by the core. This is a larger task that I'm not sure how to do today.

I could move this logic to build tracer, but that is somewhat incorrect today. The way core components work today, they manage the life cycle initialization of everything. And because `configure_tracing` is something that needs to be configured, but does not fall into the category of any other existing thing, it is being managed by the core components.

At the end of the day, I did not add this to `build_tracer` because it would be misleading to have it there. While we want with components is that they can be enabled in independently from the core tracing, adding it directly to the tracer building interface would not be correct.

Again, I'm following today's standard, and how the code is structured. With a different compartmentalization of products, this would likely live somewhere else.